test/groovy/com/sap/piper/MapUtilsTest.groovy

```diff
@@ -50,4 +50,35 @@ class MapUtilsTest {
         MapUtils.traverse(m, { s -> (s.startsWith('x')) ? "replaced" : s})
         assert m == [a: 'replaced', m: [b: 'replaced', c: 'otherString']]
     }
+
+    @Test
+    void testDeepCopy() {
+
+        List l = ['a', 'b', 'c']
+
+        def original = [
+                list: l,
+                set: (Set)['1', '2'],
+                nextLevel: [
+                    list: ['x', 'y'],
+                    duplicate: l,
+                    set: (Set)[9, 8, 7]
```

Comment:
Would `[9, 8, 7] as Set` work here?

Reply:
Yes, this would also work here. So we have several options to get a Set here rather than a list.